### PR TITLE
Add asArray option to recursively transform into arrays

### PR DESCRIPTION
**Pull request by fcastillo-es**
As you commented in #8 you wanted this functionality optional I added it disabled by default in this PR.

### CODE REVIEW

File: src/Utils/AccessTrait.php [hunk 1]
```diff
@@ -8,17 +8,47 @@
...
+			return $val;
+		}
+		elseif (is_callable([$val, 'asArray']))
+		{
+			return $val->asArray(true);
+		}
+		else
+		{
+			// Fallback for stdClass objects
+			return json_decode(json_encode($val), true);
```
Comment (Art4) [inline src/Utils/AccessTrait.php]: I think array casting with `return (array) $val;` would do the same in less time.

**Comment by Art4** [comment]
Looks good to me. :+1: But please replace the json_decode/encode line with array casting.

And some tests for `objectTransform()` would be nice. :smile:

**Comment by fcastillo-es** [comment]
An array cast doesn't work with nested stdClass, which would not be transformed to array. See this:

https://gist.github.com/fcastillo-es/03305bbbd51f5ed00b8f

That would cause failure with valid meta members. Tell me what you think about it and I'll take care of the tests.

**Comment by Art4** [comment]
You are right, I've forgotten about nested classes. Alright then.

**Comment by fcastillo-es** [comment]
Ok, I added some tests. Basically duplicating all the assertSame assertions with *fullArray* true. Also added an assertion to each integration test. I think this is enough.

By the way, I think I messed up a commit when trying to rebase and maybe I should create another pull request to have a clean history. Tell me when you can.

**Comment by Art4** [comment]
Thanks! Looks goot to me.

> By the way, I think I messed up a commit when trying to rebase

I thinks thats not a problem. You rebased your commit from #9.

**Pull request merged**